x/stake/types/msg.go

```diff
@@ -27,36 +27,53 @@ var maximumBondingRationalDenominator sdk.Int = sdk.NewInt(int64(math.Pow10(MaxB
 // MsgCreateValidator - struct for unbonding transactions
 type MsgCreateValidator struct {
 	Description
-	ValidatorAddr  sdk.Address   `json:"address"`
-	PubKey         crypto.PubKey `json:"pubkey"`
-	SelfDelegation sdk.Coin      `json:"self_delegation"`
+	DelegatorAddr sdk.Address   `json:"delegator_address"`
+	ValidatorAddr sdk.Address   `json:"validator_address"`
+	PubKey        crypto.PubKey `json:"pubkey"`
+	Delegation    sdk.Coin      `json:"delegation"`
 }
 
+// Default way to create validator. Delegator address and validator address are the same
 func NewMsgCreateValidator(validatorAddr sdk.Address, pubkey crypto.PubKey,
 	selfDelegation sdk.Coin, description Description) MsgCreateValidator {
 	return MsgCreateValidator{
-		Description:    description,
-		ValidatorAddr:  validatorAddr,
-		PubKey:         pubkey,
-		SelfDelegation: selfDelegation,
+		Description:   description,
+		DelegatorAddr: validatorAddr,
+		ValidatorAddr: validatorAddr,
+		PubKey:        pubkey,
+		Delegation:    selfDelegation,
+	}
+}
+
+// Creates validator msg by delegator address on behalf of validator address
+func NewMsgCreateValidatorOnBehalfOf(delegatorAddr, validatorAddr sdk.Address, pubkey crypto.PubKey,
+	delegation sdk.Coin, description Description) MsgCreateValidator {
+	return MsgCreateValidator{
+		Description:   description,
+		DelegatorAddr: delegatorAddr,
+		ValidatorAddr: validatorAddr,
+		PubKey:        pubkey,
+		Delegation:    delegation,
 	}
 }
 
 //nolint
 func (msg MsgCreateValidator) Type() string { return MsgType }
 func (msg MsgCreateValidator) GetSigners() []sdk.Address {
-	return []sdk.Address{msg.ValidatorAddr}
+	return []sdk.Address{msg.DelegatorAddr}
```

Comment:
We should definately also include requirement of the validators signature - you should require their permission, so you should require their signature - not sure how we actually get two folks to sign a transaction atm but we should still include it, even if it's not currently able to be used properly from the CLI

Reply:
Alright, have to remove my CLI test tho. :(

Reply:
Ready for review now